### PR TITLE
Bump to .NET 6 stable

### PR DESCRIPTION
**Pull request by mfkl**

### CODE REVIEW

**Comment by mfkl** [comment]
I'm HTTP 401 to access the CI build logs, but my guess is the samples don't build because of https://github.com/xamarin/Xamarin.Legacy.Sdk/issues/33, or did I mess up somewhere else?

**Comment by jonathanpeppers** [comment]
There is some failure in the iOS workload:
```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: The "GenerateBuildSessionId" task failed unexpectedly. [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: System.TypeLoadException: Method 'Trace' in type 'NullTracer' from assembly 'Xamarin.iOS.Tasks, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation. [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager..ctor() [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager.get_Instance() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\BuildTracerManager.cs:line 23 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.SetupTracing() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 49 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.Initialize(String version) in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 15 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.Tasks.GenerateBuildSessionId.Execute() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\Tasks\GenerateBuildSessionId.cs:line 26 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: The "GenerateBuildSessionId" task failed unexpectedly. [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: System.TypeLoadException: Method 'Trace' in type 'NullTracer' from assembly 'Xamarin.iOS.Tasks, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation. [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager..ctor() [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager.get_Instance() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\BuildTracerManager.cs:line 23 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.SetupTracing() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 49 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.Initialize(String version) in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 15 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.Tasks.GenerateBuildSessionId.Execute() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\Tasks\GenerateBuildSessionId.cs:line 26 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
```

We're using Xamarin.Legacy.Sdk + .NET 6 GA, in [AndroidX](https://github.com/xamarin/AndroidX) and Google Play Services bindings...

Maybe there is something I can do to work around the iOS issue.

File: src/Xamarin.Legacy.Sdk/Sdk/Sdk.props [hunk 1]
```diff
@@ -3,5 +3,21 @@
...
+  -->
+  <Target Name="_GetRestoreSettingsPerFramework" Returns="@(_RestoreSettingsPerFramework)">
+    <ItemGroup>
+      <_RestoreSettingsPerFramework Include="$([System.Guid]::NewGuid())">
+        <RestoreAdditionalProjectSources>$(RestoreAdditionalProjectSources)</RestoreAdditionalProjectSources>
+        <RestoreAdditionalProjectFallbackFolders>$(RestoreAdditionalProjectFallbackFolders)</RestoreAdditionalProjectFallbackFolders>
+        <RestoreAdditionalProjectFallbackFoldersExcludes>$(RestoreAdditionalProjectFallbackFoldersExcludes)</RestoreAdditionalProjectFallbackFoldersExcludes>
+      </_RestoreSettingsPerFramework>
+    </ItemGroup>
+  </Target>
```
Comment (jonathanpeppers) [inline src/Xamarin.Legacy.Sdk/Sdk/Sdk.props]: @mfkl adding this solved one issue, were you getting errors about this target missing?

After adding this, I hit a new missing target, which is too large to replicate in this project:

https://github.com/NuGet/NuGet.Client/blob/914cacea77660b8758014912ee127c82469ac1d9/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets#L856-L863

I think these were very late changes to improve no-op restore performance in NuGet in .NET 6.

Need to think on this one...

Comment (jonathanpeppers) [reply src/Xamarin.Legacy.Sdk/Sdk/Sdk.props]: Ok remembering now... This needs to be updated for VS 2022:

https://github.com/xamarin/Xamarin.Legacy.Sdk/blob/4ba5930273eb5bd64b142cfdc574626c68ad41ab/src/Xamarin.Legacy.Sdk/Sdk/Xamarin.VsInstallRoot.targets#L6-L7

**Comment by jonathanpeppers** [comment]
This is now hitting a new iOS error:
```
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: The "GenerateBuildSessionId" task failed unexpectedly. [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: System.TypeLoadException: Method 'Trace' in type 'NullTracer' from assembly 'Xamarin.iOS.Tasks, Version=15.2.0.17, Culture=neutral, PublicKeyToken=84e04ff9cfb79065' does not have an implementation. [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager..ctor() [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager.get_Instance() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\BuildTracerManager.cs:line 23 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.SetupTracing() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 49 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.Initialize(String version) in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 15 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.Tasks.GenerateBuildSessionId.Execute() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\Tasks\GenerateBuildSessionId.cs:line 26 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: The "GenerateBuildSessionId" task failed unexpectedly. [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: System.TypeLoadException: Method 'Trace' in type 'NullTracer' from assembly 'Xamarin.iOS.Tasks, Version=15.2.0.17, Culture=neutral, PublicKeyToken=84e04ff9cfb79065' does not have an implementation. [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager..ctor() [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager.get_Instance() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\BuildTracerManager.cs:line 23 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.SetupTracing() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 49 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.Initialize(String version) in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 15 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.Tasks.GenerateBuildSessionId.Execute() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\Tasks\GenerateBuildSessionId.cs:line 26 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
```

**Comment by mfkl** [comment]
Good job!

**Pull request merged**